include/range/v3/view/drop_last.hpp

```diff
@@ -0,0 +1,237 @@
+
+                adaptor(CRng &rng, difference_t n)
+                {
+                    probe = ranges::next(ranges::begin(rng), n, ranges::end(rng));
+                }
+
+                void next(iterator_t<CRng> &it)
+                {
+                    ++it;
+                    ++probe;
+                }
+            };
+
+            struct sentinel_adaptor :  adaptor_base
+            {
+                template<typename I, bool is_const, typename S>
+                bool empty(I const &, adaptor<is_const> const &ia, S const &s) const
+                {
+                    return ia.probe == s;
+                }
+            };
+
+            adaptor<false> begin_adaptor() { return {this->base(), n}; }
+            sentinel_adaptor end_adaptor() { return {}; }
+
+            CONCEPT_REQUIRES(Range<Rng const>())
+            adaptor<true>  begin_adaptor() const { return {this->base(), n}; }
+
+            CONCEPT_REQUIRES(Range<Rng const>())
+            sentinel_adaptor end_adaptor() const { return {}; }
+
+            template<class T>
+            size_t get_size(T &&rng) const
+            {
+                const size_t initial_size = ranges::size(rng);
+                const size_t n = static_cast<size_t>(this->n);
+
+                return initial_size > n
+                    ? initial_size - n
+                    : 0;
+            }
+        public:
+            drop_last_view() = default;
+            drop_last_view(Rng rng, difference_t n)
+              : drop_last_view::view_adaptor(std::move(rng))
+              , n(n)
+            {}
+
+            CONCEPT_REQUIRES(SizedRange<Rng>())
+            size_t size()
+            {
+                return get_size(this->base());
+            }
+            CONCEPT_REQUIRES(SizedRange<Rng const>())
+            size_t size() const
+            {
+                return get_size(this->base());
+            }
+        };
+
+        namespace view
+        {
+            struct drop_last_fn
+            {
+            private:
+                friend view_access;
+
+                template<typename Int, CONCEPT_REQUIRES_(Integral<Int>())>
+                static auto bind(drop_last_fn drop_last, Int n)
+                RANGES_DECLTYPE_AUTO_RETURN
+                (
+                    make_pipeable(std::bind(drop_last, std::placeholders::_1, n))
+                )
+
+                template<typename Rng>
+                static drop_last_view<all_t<Rng>>
+                invoke_(Rng &&rng, range_difference_type_t<Rng> n)
+                {
+                    return {all(static_cast<Rng&&>(rng)), n};
+                }
+            public:
+                template<typename Rng,
+                    CONCEPT_REQUIRES_(ForwardRange<Rng>())>
+                auto operator()(Rng &&rng, range_difference_type_t<Rng> n) const
+                RANGES_DECLTYPE_AUTO_RETURN
+                (
+                    drop_last_fn::invoke_(static_cast<Rng&&>(rng), n)
```

Comment:
Why not just return `drop_last_view<all_t<Rng>>{all(static_cast<Rng&&>(rng)), n}` ?

Reply:
Instead of `drop_last_fn::invoke_(static_cast<Rng&&>(rng), n)` ?